Newsletters Error (BeautifulSoup Helper)

stryker2k2 opened the pull request:
### Description:
- Newsletters (along with dnsdumpster and viewdns) utilize the newly created BeautifulSoup Helper. But, there are instances where BeautifulSoup Helper would return "False" and the modules were not created to handle that boolean. I have gone through each module and added a Try/Catch to them. To prove that the try/catch would work as intended, I tested it by forcing the helper.beautifulsoup to always return False then I ran a scan.

### Ticket/Issue
- https://github.com/blacklanternsecurity/bbot/issues/1138

### To Review:
- [x] Run BBOT normally with these modules using the following command
    - `bbot -m httpx newsletters dnsdumpter viewdns -t example.com`
    - Expect everything to work fine (with results from modules, if target permits)
- [x] Run BBOT after setting the BeautifulSoup Helper to return False
    - bbot/core/helpers/web.py:623
        - return ~~soup~~ False
    - `bbot -m httpx newsletters dnsdumpter viewdns -t example.com`
    - Expect everything to work fine (with NO RESULTS from selected modules)
    - Expect no crashes (as seen in #1138)

### Notes:
- This fixes the issue and will prevent BBOT from crashing... but I still wonder why BeautifulSoup didn't work as intended in https://github.com/blacklanternsecurity/bbot/issues/1138

stryker2k2 committed:
Added try/catch for modules that utilize BeautifulSoup

stryker2k2 committed:
Added try/catch AND ran black reformatter

TheTechromancer commented on bbot/modules/newsletters.py:
```diff
@@ -33,10 +33,17 @@ def find_type(self, soup):
+            # If BeautifulSoup Fails, we will return back to the calling function with no action taken
+            except Exception:
+                self.debug(f"BeautifulSoup returned False")
+                pass
```
There's no reason to wrap in a blanket try/except here since you're already checking for falsiness.

stryker2k2 replied:
Roger that - change has been made.

TheTechromancer commented on bbot/modules/dnsdumpster.py:
```diff
@@ -24,7 +24,12 @@ async def query(self, domain):
         else:
             self.debug(f'Valid response code "{status_code}" from DNSDumpster')
-        html = self.helpers.beautifulsoup(res1.content, "html.parser")
+        try:
+            html = self.helpers.beautifulsoup(res1.content, "html.parser")
+        # If BeautifulSoup Fails, we will return back to the calling function with no action taken
+        except Exception:
+            self.debug(f"BeautifulSoup returned False")
+            pass
```
Again, the try/except will not catch the error because it's being caught already in the helper. Something like `if html is False: return` would suffice.

stryker2k2 replied:
Roger that - change has been made.

stryker2k2 committed:
replaced try/catch with if/else statements

TheTechromancer commented:
This should good to merge once the last instance is fixed in dnsdumpster.py.

stryker2k2 committed:
Merge branch 'dev' into newsletters_error

stryker2k2 committed:
if/else blocks in DNSDumpster, return ret[], and print self.verbose

stryker2k2 committed:
Fixed 'KeyError: body' bug

stryker2k2 commented:
@TheTechromancer - it is ready for review and merge 😄 I've fixed the last remaining instance in `dnsdumpster`, pulled in `dev`, added in the fix for the new issue that amiremami posted about a few hours ago about the `KeyError: 'body'`
